atomix/cluster/src/test/java/io/atomix/raft/partition/PartitionDistributorTest.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft.partition;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.cluster.MemberId;
+import io.atomix.primitive.partition.PartitionId;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Stream;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class PartitionDistributorTest {
+
+  @ParameterizedTest(name = "[{index}] {0} {1} {2}")
+  @MethodSource("clusterConfig")
+  void shouldAssignPrioritiesCorrectly(
+      final int clusterSize,
+      final int partitionCount,
+      final int replicationFactor,
+      final int[][] expected) {
+    // when
+    final var partitionMetadata =
+        new PartitionDistributor()
+            .generatePartitionDistribution(
+                getMembers(clusterSize), getSortedPartitionIds(partitionCount), replicationFactor);
+
+    // then
+    partitionMetadata.forEach(
+        metadata -> {
+          final int partitionId = metadata.id().id();
+          metadata
+              .members()
+              .forEach(
+                  member -> {
+                    final int nodeId = Integer.parseInt(member.id());
+                    assertThat(metadata.getPriority(member))
+                        .describedAs(
+                            "Priority calculated wrong for nodeId %d, partition %d. Observed metadata is %s",
```

Comment:
Might make sense to print also the expected :)

Reply:
That would be printed by default. This is how it looks like:
```
org.opentest4j.AssertionFailedError: [Priority calculated wrong for nodeId 1, partition 3. Observed metadata is PartitionMetadata{id=PartitionId{id=3, group=test}, members=[2, 0, 1], priority={1=1, 2=3, 0=2}, targetPriority=3}] 
expected: 0
 but was: 1
Expected :0
Actual   :1
```